Sources/TuistDependencies/Carthage/Utils/CarthageGraphGenerator.swift

```diff
@@ -41,8 +41,7 @@ public final class CarthageGraphGenerator: CarthageGraphGenerating {
                         frameworkName,
                     ])
 
-                let architectures: Set<BinaryArchitecture> = Set(products.flatMap { $0.architectures })
-                return .xcframework(name: dependencyName, path: path, architectures: architectures)
+                return [.xcframework(path: .init(path.pathString))]
```

Comment:
@laxmorek Given the TargetDependency does not require the architecture, we can get rid of all the logic to compute it from the cartfile, isn't it?

Reply:
Hmmm. I have found something like [this](https://github.com/tuist/tuist/blob/main/Sources/TuistGraph/Models/XCFrameworkInfoPlist.swift). I can see it is used [here](https://github.com/tuist/tuist/blob/c828fe563d66662902d8f7c3838901f258acb347/Sources/TuistCore/Graph/GraphDependencyReference.swift#L6).

@fortmarek @kwridan Can you help us? 🙄

Reply:
Not sure about that, but I assumed that it would have been retrieved by tuist looking at the content of the framework, the same way it works when defining a TargetDependency.xcframework dependency

Reply:
yeah, that plist is obtained via `XCFrameworkMetadataProvider` which just needs path to the framework as an input. Considering `TargetDependency.xcframework` does not have the architecture property, we should be fine by encoding the graph without it

Reply:
however, I am not sure if I follow `.init(path.pathString)`? Firstly, I'd use the type name instead of `init`. And if the type is `AbsolutePath`, is it even necessary? Or is the type something different?

Reply:
@fortmarek I think it's manifest path type `ProjectDescription.Path` _(with has a few additional features like `.relativeToRoot`)_, manifests don't usually contain absolute paths (as they aren't portable between different setups) but I believe it will support them if specified.

Reply:
ahh, that makes absolute sense, thanks! Using the type name instead of `.init` would then help understanding it 🙂